### ISSUE TITLE
Schema errors are not rendered correctly after switch to iter_errors

### ISSUE THREAD

**Issue by dudicoco** [sirosen/check-jsonschema#24]
Given the following configuration:

values.json:
```json
{
  "$schema": "http://json-schema.org/draft-07/schema",
  "$defs": {
    "test": {
      "type": "string"
    }
  }
}
```

test.json:
```json
{
  "$schema": "http://json-schema.org/draft-07/schema",
  "type": "object",
  "required": ["test"],
  "properties": {
    "test": {
      "$ref": "./values.json#/$defs/test"
    }
  }
}
```

test.yaml:
```yaml
test:
  foo: bar
```

Getting the following error:
```sh
Schema validation errors were encountered.
Traceback (most recent call last):
  File "/Users/xxxx/.cache/pre-commit/repo7ycxoqhn/py_env-python3.9/bin/check-jsonschema", line 8, in <module>
    sys.exit(main())
  File "/Users/xxxx/.cache/pre-commit/repo7ycxoqhn/py_env-python3.9/lib/python3.9/site-packages/check_jsonschema/__init__.py", line 16, in main
    checker.run()
  File "/Users/xxxx/.cache/pre-commit/repo7ycxoqhn/py_env-python3.9/lib/python3.9/site-packages/check_jsonschema/checker.py", line 106, in run
    path = [str(x) for x in err.path] or ["<root>"]
AttributeError: 'list' object has no attribute 'path'
```

**Comment by sirosen** [comment]
There's no issue with local refs here. That data is not valid under the schema you have given.

The error is being misreported, which is interesting, however. I didn't correctly adjust things when I started using `iter_errors` to get all errors for a schema.

**Comment by sirosen** [comment]
I've just tested a fix for this. The error will render correctly in the next release:
```bash
$ check-jsonschema --schemafile test.json test.yaml
Schema validation errors were encountered.
  test.yaml::test: {'foo': 'bar'} is not of type 'string'
```

Thanks for reporting! I'll close this when I have the fix out.

**Comment by sirosen** [comment]
Resolved in v0.7.1

**Comment by dudicoco** [comment]
Thanks @sirosen! I can confirm that the issue has been resolved.
